File: product_sdk/update_product_request.go
package productsdk

import (
	"bytes"
	"encoding/json"
	"io"
	"log"
)

type UpdateProductRequest struct {
	Name        *string  `json:"name"`
	Description *string  `json:"description"`
	Price       *float64 `json:"price"`
	Quantity    *int     `json:"quantity" validate:"gt=0"`
}

func (u *UpdateProductRequest) ToJson() io.Reader {
	buffer := new(bytes.Buffer)
	log.Println(u)
	json.NewEncoder(buffer).Encode(u)

	return buffer
}

File: frontend/shoppie_admin/lib/features/products/models/update_product_request.dart
import 'package:json_annotation/json_annotation.dart';

part "update_product_request.g.dart";

@JsonSerializable()
class UpdateProductRequest {
  final String id;
  final String name;
  final String description;
  final double price;
  final int quantity;

  UpdateProductRequest({
    required this.description,
    required this.id,
    required this.name,
    required this.price,
    required this.quantity,
  });

  factory UpdateProductRequest.fromJson(Map<String, dynamic> json) =>
      _$UpdateProductRequestFromJson(json);

  Map<String, dynamic> toJson() => _$UpdateProductRequestToJson(this);
}
